firmware/projects/FrontController/inc/app.h

```diff
@@ -0,0 +1,107 @@
+/// @author Blake Freer
+/// @date 2024-02-27
+
+#pragma once
+
+#include <cstdint>
+
+#include "shared/periph/adc.h"
+#include "shared/periph/gpio.h"
+
+class Pedal {
+public:
+    Pedal(shared::periph::ADCInput& adc) : adc_(adc) {}
+
+    uint16_t Update() {
+        /// @todo Map adc value to 0-100% range for standardization.
+        position_ = uint16_t(adc_.Read());
+        return GetPosition();
+    }
+
+    /**
+     * @brief Get the position from the last `Update()` call.
+     */
+    inline uint16_t GetPosition() const {
+        return position_;
+    }
+
+private:
+    uint16_t position_ = 0;
+    shared::periph::ADCInput& adc_;
+};
+
+class SteeringWheel {
+public:
+    SteeringWheel(shared::periph::ADCInput& adc) : adc_(adc) {}
+
+    uint16_t Update() {
+        /// @todo Map to range [-1, 1] or [-180 deg, +180 deg] etc
+        position_ = uint16_t(adc_.Read());
+        return GetPosition();
+    }
+
+    /**
+     * @brief Get the position from the last `Update()` call.
+     */
+    inline uint16_t GetPosition() const {
+        return position_;
+    }
+
+private:
+    uint16_t position_ = 0;
+    shared::periph::ADCInput& adc_;
+};
+
+class Speaker {
+public:
+    bool state;
```

Comment:
Should we modify this through an accessor like `SetState` to more closely match the style of `SteeringWheel` for example

Reply:
depends on how we want our Update to work. If SetState updates the GPIO then yes I agree, but if we require a separate Update function (for example, to update all GPIO in the same task), then I don't see the benefit of hiding a boolean variable. This is a slightly different setup than SteeringWheel since Speaker / BrakeLight are outputs, not inputs.